e2e/test/provisioning/ProvisioningE2ETests.cs

```diff
@@ -1228,6 +1188,49 @@ public static bool ImplementsWebProxy(Client.TransportType transportProtocol)
             throw new NotSupportedException($"Unknown transport: '{transportProtocol}'.");
```

Comment:
Unrelated to this PR, but I thought it was cool so I thought I'd share:
Intellisense recommends the following for `ImplementsWebProxy`

```c#
return transportProtocol switch
    {
        Client.TransportType.Mqtt_WebSocket_Only or Client.TransportType.Amqp_WebSocket_Only => true,
        _ => false,
    };
```

Reply:
Oh wow that is cool